### PR TITLE
=per Fix flaky ThrowExceptionStrategyPersistentActorBoundedStashingSpec

### PR DESCRIPTION
**Pull request by bantonsson**
This test triggered intermittent failures in some internal builds. It had a small race that sometime manifested on the slow EC2 nodes running it.

Could easily trigger the behaviour by adding some sleeps in message sending and mailbox cleanup.

### CODE REVIEW

**Comment by bantonsson** [comment]
Note to self. Reopen the PR before force pushing, or else it's not possible.

File: akka-persistence/src/test/scala/akka/persistence/PersistentActorBoundedStashingSpec.scala [hunk 1]
```diff
@@ -90,11 +101,23 @@ class ThrowExceptionStrategyPersistentActorBoundedStashingSpec
...
+      val (stashed, nonStashed) = messages.partition {
+        case DeadLetter(Cmd(i: Int), `testActor`, `persistentActor`) ⇒ i <= capacity
+      }
+
+      checkMessagesAreInOrder(stashed, 1)
+      //since we don't know exactly which message is the last one going into the mailbox and which one is the
+      //first one going directly to dead letters, we can't assume an ordering and need to sort the messages
+      checkMessagesAreInOrder(nonStashed.sortBy {
+        case DeadLetter(Cmd(i: Int), _, _) ⇒ i
+      }, capacity + 2)
```
Comment (patriknw) [inline akka-persistence/src/test/scala/akka/persistence/PersistentActorBoundedStashingSpec.scala]: Does this test take into account that message may be lost (not sent to deadLetters) during actor stopping?

Comment (bantonsson) [inline akka-persistence/src/test/scala/akka/persistence/PersistentActorBoundedStashingSpec.scala]: @patriknw Could you elaborate on when the message may be lost? Are we talking about the very small window between `actor.swapMailbox` and `mailBox.becomeClosed` in `unregister`, or is there something else? And no, it assumes that part works fine.

Comment (patriknw) [inline akka-persistence/src/test/scala/akka/persistence/PersistentActorBoundedStashingSpec.scala]: yes, it's the small window I'm thinking about

Comment (bantonsson) [inline akka-persistence/src/test/scala/akka/persistence/PersistentActorBoundedStashingSpec.scala]: Yup, I'm happily ignoring that one. Do you want me to update it to take that into account?

Comment (patriknw) [inline akka-persistence/src/test/scala/akka/persistence/PersistentActorBoundedStashingSpec.scala]: Is it possible to simplify the test, I'm not sure what it does.
We know exactly how many messages that are required for the stash overflow to happen? can we just verify that the actor is stopped after that? this checking of deadletters is not really a concern of this test, is it?

Comment (bantonsson) [inline akka-persistence/src/test/scala/akka/persistence/PersistentActorBoundedStashingSpec.scala]: That is an excellent suggestion. I didn't put too much thought into what the test was doing with the extra dead letters. Will simplify it.

Comment (patriknw) [inline akka-persistence/src/test/scala/akka/persistence/PersistentActorBoundedStashingSpec.scala]: thanks a lot!

**Comment by bantonsson** [comment]
Simplified test after discussion with @patriknw

**Comment by patriknw** [comment]
LGTM, thanks

**Comment by bantonsson** [comment]
So anybody else has any opinions, or can I push Da Button?

Push Da Button PLZ?

**Pull request merged**